src/ScribuntoLuaLibrary.php

```diff
@@ -93,4 +106,118 @@ public function getPropertyType( $propertyName = null ) {
 		return array( $property->findPropertyTypeID() );
 	}
 
+	/**
+	 * This mirrors the functionality of the parser function #info and makes it available to lua.
+	 *
+	 * @param string	$text	text to show inside the info popup
+	 * @param string    $icon   identifier for the icon to use
+	 *
+	 * @uses \SMW\ParserFunctionFactory::__construct
+	 *
+	 * @return string[]
+	 */
+	public function info( $text, $icon='info' )
+	{
+		$parser = $this->getEngine()->getParser();
+
+		# do some parameter processing
+		if ( !trim($text) || ! is_string($text) ) {
+			# no info-text present, or wrong type abort
+			return null;
+		}
+		$infoText = trim($text);
+
+		# check if icon is set and valid
+		if ( !is_string($icon) || !in_array($icon, ['note', 'warning']) ) {
+			$icon = 'info';
+		}
+
+		# the actual info message is easy to create:
+		$result = smwfEncodeMessages(
+			array( $infoText ),
+			$icon,
+			' <!--br-->',
+			false // No escaping.
+		);
+
+		# to have all necessary data commited to output, use SMWOutputs::commitToParser()
+		SMWOutputs::commitToParser( $parser );
+
+		# result may be an array with probably 'noparse' set to false. check to be sure
+		$noParse = ( is_array($result) && isset($result['noparse']) ) ? $result['noparse'] : true;
+		$result = is_array($result) ? $result[0] : $result;
+
+		if ( ! $noParse ) {
+			$result = $parser->recursiveTagParseFully( $result );
+		}
+
+		return array( $result );
+	}
+
+	/**
+	 * This mirrors the functionality of the parser function #set and makes it available in lua.
+	 *
+	 * @param string|array	$parameters	parameters passed from lua, string or array depending on call
+	 *
+	 * @uses \SMW\ParserFunctionFactory::__construct, ParameterProcessorFactory::newFromArray
+	 *
+	 * @return null|array|array[]
+	 */
+	public function set( $parameters )
```

Comment:
These `info` and `set` functions can go into their own classes. As is, the code in this class is not very cohesive, and it turns into a dump of "everything lua related".

Perhaps named `SMW\Scribunto\SetFunction->handleCall( $parameters )`. @mwjames thoughts? As far as I can tell this code does not actually bind to Scribunto, so that part of the name is a bit odd.